File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/Scene/SceneEntityBase/SceneEntityMovement.cs
namespace SDK.Lib
{
    /**
     * @brief 移动类型
     */
    public class SceneEntityMovement : GObject, IDispatchObject
    {
        protected SceneEntityBase mEntity;          // 关联的实体

        public SceneEntityMovement(SceneEntityBase entity)
        {
            mTypeId = "SceneEntityMovement";
            this.mEntity = entity;
        }

        virtual public void init()
        {

        }

        virtual public void dispose()
        {
            this.mEntity = null;
        }

        virtual public void onPutInPool()
        {
            
        }

        virtual public void onTick(float delta, TickMode tickMode)
        {

        }
    }
}

File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/Scene/SceneEntityBase/SceneEntityMovement.cpp
#include "MyProject.h"
#include "SceneEntityMovement.h"
#include "MClassFactory.h"

MY_BEGIN_NAMESPACE(MyNS)

M_IMPLEMENT_AND_REGISTER_CLASS(SceneEntityMovement, GObject)

SceneEntityMovement::SceneEntityMovement(SceneEntityBase* entity)
{
	this->mTypeId = "SceneEntityMovement";

	this->mEntity = entity;
}

void SceneEntityMovement::init()
{

}

void SceneEntityMovement::dispose()
{
	this->mEntity = nullptr;
}

void SceneEntityMovement::onTick(float delta, TickMode tickMode)
{

}
	
MY_END_NAMESPACE